text_extensions_for_pandas/transformers/test_tokenization.py

```diff
@@ -0,0 +1,73 @@
+#
+#  Copyright (c) 2020 IBM Corp.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+import unittest
+
+import pandas as pd
+from transformers import BertTokenizerFast
+
+from text_extensions_for_pandas.transformers import tokenization
+
+
+class TestTokenize(unittest.TestCase):
+
+    def setUp(self):
+        # Ensure that diffs are consistent
+        pd.set_option("display.max_columns", 250)
+
+    def tearDown(self):
+        pd.reset_option("display.max_columns")
+
+    def test_make_bert_tokens(self):
+        text = "the quick brown fox blumped over a blazy bog."
+        tokenizer = BertTokenizerFast.from_pretrained('bert-base-uncased', add_special_tokens=True)
+        bert_toks = tokenization.make_bert_tokens(text, tokenizer)
+
+        self.assertIsInstance(bert_toks, pd.DataFrame)
+        self.assertEqual(len(bert_toks.columns), 7)
+
+        # verify input id is correct
+        self.assertTrue('input_id' in bert_toks.columns)
+        input_id = bert_toks['input_id']
+        self.assertEqual(input_id.iloc[0], 101)  # docstart
+        self.assertEqual(input_id.iloc[-1], 102)  # docend
+        self.assertEqual(len(set(input_id)), len(input_id))
+
+        # verify special tokens mask
+        self.assertTrue('special_tokens_mask' in bert_toks.columns)
+        special_tokens_mask = bert_toks['special_tokens_mask']
+        self.assertEqual(special_tokens_mask.iloc[0], True)
+        self.assertTrue(not any(special_tokens_mask.iloc[1:-1]))
+        self.assertEqual(special_tokens_mask.iloc[-1], True)
+
+        # verify char span array
+        self.assertTrue('char_span' in bert_toks.columns)
+        char_span = bert_toks['char_span']
+        span = char_span.iloc[0]
+        self.assertEqual(span.begin, span.end)  # docstart special token
+        span = char_span.iloc[-1]
+        self.assertEqual(span.begin, span.end)  # docend special token
+        expected = ["the", "quick", "brown", "fox", "over", "a", "bog"]
```

Comment:
I wasn't sure if those would be tokenized consistently or it might depend on some setting since they aren't words. Should I check the output as is or use real words instead?

Reply:
I used real words, we can test fancier things later